src/aosm/azext_aosm/common/artifact.py

```diff
@@ -82,12 +83,26 @@ def _manifest_credentials(
         aosm_client: HybridNetworkManagementClient,
     ) -> MutableMapping[str, Any]:
         """Gets the details for uploading the artifacts in the manifest."""
-        return aosm_client.artifact_manifests.list_credential(
-            resource_group_name=config.publisherResourceGroupName,
-            publisher_name=config.publisherName,
-            artifact_store_name=config.acrArtifactStoreName,
-            artifact_manifest_name=config.acrManifestName,
-        ).as_dict()
+        retries = 0
+        # This retry logic is to handle the ServiceResponseError that is hit in the integration tests.
+        # This error is not hit when running the cli normally because the CLI framework automatically retries,
+        # the testing framework does not support automatic retries.
+        while retries < 2:

```

Comment:
Couple of questions: 

- Are automatic retries going to race/conflict with these ones? I'm not sure what would happen if they did - maybe it's fine.
- Why aren't these retries in the test code if it's a limitation in the test framework?

Reply:
- The automatic retires should not race with this one because the ServiceResponseError is only raised when running in a test. 
- I have discussed with the rest of the team about fixing in the testing code and we decided that it would be complicated to figure out (if even possible) so we should just fix it in the main code and chase the CLI team about this problem